libbpf-tools/tcppktlat.c

```diff
@@ -0,0 +1,262 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+// Copyright (c) 2023 Wenbo Zhang
+#include <argp.h>
+#include <signal.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <time.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <sys/resource.h>
+
+#include <bpf/bpf.h>
+#include <bpf/libbpf.h>
+#include "tcppktlat.h"
+#include "tcppktlat.skel.h"
+#include "compat.h"
+#include "trace_helpers.h"
+
+static struct env {
+	pid_t pid;
+	pid_t tid;
+	__u64 min_us;
+	__u16 lport;
+	__u16 rport;
+	bool timestamp;
+	bool verbose;
+} env = {};
+
+static volatile sig_atomic_t exiting = 0;
```

Comment:
Any reason why we need 'volatile' keyword here?

Reply:
This is mentioned in the [signal manpage](https://man7.org/linux/man-pages/man3/signal.3p.html).